modules/web-console/frontend/app/configuration/generator/generator/JavaTransformer.service.js

```diff
@@ -990,6 +874,114 @@ export default class IgniteJavaTransformer extends AbstractTransformer {
         return beans;
     }
 
+    /**
+     * @returns {Set.<String>}
+     */
+    static collectConfigurationImports(cfg, imports = new Set()) {
```

Comment:
This function is not pure. I believe pure functions are easier to reason about. Please adjust the function signature so it accepts config and outputs imports.

Reply:
The function is purified.